Framework/Core/src/DataProcessingDevice.cxx

```diff
@@ -1155,26 +1155,54 @@ bool DataProcessingDevice::tryDispatchComputation(DataProcessorContext& context,
   };
 
   //
+  // the number of parts per input depends on the number of added PartRef object
+  // and the number of payloads in these objects. ProcessingContext provdes data
+  // through the input span where every payload is provided through a DataRef
+  // object. The part index within the input needs to be translated into a 2 dim
+  // index to access the payload message i from PartRef object j and we generate
+  // a consecutive map of these tuples
+  std::vector<std::vector<std::pair<size_t, size_t>>> partRefMaps;
```

Comment:
By isolating the individual parts of #7110, its now verified that the unit test failure of QualityControl/testCheckWorkflow is triggered by introducing this extra variable and using it in the lambdas. Since this variable is defined at the functions' scope like `currentSetOfInputs` and it is captured by reference we can rule out a simple out-of-scope volatile object.

This needs to be further investigate to exclude other errors. Here we have solved the problem by adding the ref index to the MessageSet. Adding this commit solves the unit test failure.

Reply:
found the problem. This index can go out of sync with the vector of message sets in the loop for processing the list of ready-actions. In this loop, the message set can now be cleared by early forwarding. The index has now been incorporated into `MessageSet`.